northstar/src/runtime/mount.rs

```diff
@@ -238,23 +260,33 @@ pub async fn verity_setup(
 async fn setup_and_mount(
     config: &Config,
     name: &str,
-    npk_reader: &ArchiveReader,
+    npk_path: &Path,
     root: &Path,
     use_verity: bool,
 ) -> Result<PathBuf, Error> {
-    let fs_type = npk_reader.fs_type().map_err(Error::Npk)?;
-    let mut fs = fs::File::open(&npk_reader.path)
+    // open NPK synchronously
+    let (verity, hashes, fsimg_offset, fsimg_size) = tokio::task::block_in_place(|| {
+        let file = File::open(npk_path)
+            .map_err(|e| npk::npk::Error::FileOperation(format!("Failed to open NPK: {}", e)))?;
+        let mut npk = Npk::new(file)?;
+        Ok((
+            npk.verity_header()?,
+            npk.hashes()?,
+            npk.fsimg_offset()?,
+            npk.fsimg_size()?,
+        ))
+    })
+    .map_err(Error::Npk)?;
+
+    let mut fs = fs::File::open(npk_path)
```

Comment:
This needs a `block_in_place`

Reply:
Do you mean this call? Isn't it async already?
let mut npk_file = fs::File::open(npk_path).await ...